.github/workflows/onPushToMain.yml

```diff
@@ -0,0 +1,10 @@
+name: version, tag and github release
```

Comment:
How did we test all these flows? Or do we need to test when we do the next release

Reply:
I couldn't find any easy way to test the release flows without actually releasing so I think it'll be testing when we do next release